pkg/util/validation/limits.go

```diff
@@ -885,3 +917,93 @@ func EnabledByAnyTenant(tenantIDs []string, f func(string) bool) bool {
 	}
 	return false
 }
+
+// RegisterExtensions registers the extensions type with given name
+// and returns a function to get the extensions from a *Limits instance.
```

Comment:
to get a pointer to extensions

Reply:
Updated the comment here: https://github.com/grafana/mimir/pull/4397/commits/c81aa48ecdbbcbe5fff451620a41211da1f67152

But anyway, the good part here is that it will be ensured by the compiler.